app/jobs/health/process_enrollment_changes_job.rb

```diff
@@ -122,26 +145,18 @@ def disenroll_patient(transaction, referral)
     end
 
     def update_patient_referrals(patient, transaction)
-      updates = {
-        first_name: Health::Enrollment.first_name(transaction),
-        last_name: Health::Enrollment.last_name(transaction),
-        birthdate: Health::Enrollment.DOB(transaction),
-        ssn: Health::Enrollment.SSN(transaction),
-        medicaid_id: Health::Enrollment.subscriber_id(transaction),
-        enrollment_start_date: Health::Enrollment.enrollment_date(transaction),
-      }
+      updates = referral_data(transaction)
+      current_referral = patient.patient_referral
+      current_referral.assign_attributes(updates)
 
-      health_enrollment_aco_pid_sl = Health::Enrollment.aco_pid_sl(transaction)
-      if health_enrollment_aco_pid_sl
-        pid_sl = Health::AccountableCareOrganization.split_pid_sl(health_enrollment_aco_pid_sl)
-        aco = Health::AccountableCareOrganization.active.find_by(
-          mco_pid: pid_sl[:pid],
-          mco_sl: pid_sl[:sl],
-        )
-        updates[:aco] = aco if aco.present?
-      end
+      return unless current_referral.changed?
 
-      Health::PatientReferral.create_referral(patient, updates)
+      if current_referral.enrollment_start_date_changed? || current_referral.aco_changed?
```

Comment:
Let's move this to `Health::PatientReferral.should_create_referral?` 
It will make the logic easier to adjust down the road.